### PR TITLE
improvement: Remove broker dependencies from extensions management mo…

### PR DESCRIPTION
**Pull request by dominikriemer**
…dule

  
  

### Purpose

Remove dependencies on specific broker implementations from the `extensions-management` module since these are dynamically defined by the extension module itself.
When adding extensions which only support a certain messaging system, non-needed dependencies should not be part of the module.

### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

File: streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/preprocessing/elements/SendToBrokerAdapterSink.java [hunk 1]
```diff
@@ -86,9 +87,12 @@ protected void sendToBroker(byte[] event) throws RuntimeException {
     producer.publish(event);
   }
 
-  protected abstract EventProducer makeProducer(T protocol);
-
-  public abstract void modifyProtocolForDebugging(T protocol);
+  public void modifyProtocolForDebugging(TransportProtocol protocol) {
+    protocol.setBrokerHostname("localhost");
+    if (protocol instanceof KafkaTransportProtocol) {
+      ((KafkaTransportProtocol) protocol).setKafkaPort(9094);
```
Comment (bossenti) [inline streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/preprocessing/elements/SendToBrokerAdapterSink.java:93]: The kafka port might not always be `9094` right?

File: streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/preprocessing/elements/SendToBrokerAdapterSink.java [hunk 2]
```diff
@@ -86,9 +87,12 @@ protected void sendToBroker(byte[] event) throws RuntimeException {
     producer.publish(event);
   }
 
-  protected abstract EventProducer makeProducer(T protocol);
-
-  public abstract void modifyProtocolForDebugging(T protocol);
+  public void modifyProtocolForDebugging(TransportProtocol protocol) {
+    protocol.setBrokerHostname("localhost");
```
Comment (bossenti) [inline streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/preprocessing/elements/SendToBrokerAdapterSink.java:91]: Does this also work in a Kubernetes deployment?

File: streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/AdapterPipelineGenerator.java [hunk 3]
```diff
@@ -51,31 +39,8 @@ public AdapterPipeline generatePipeline(AdapterDescription adapterDescription) {
...
-  private boolean isPrioritized(SpProtocol prioritizedProtocol,
-                                Class<?> protocolClass) {
-    return prioritizedProtocol.getProtocolClass().equals(protocolClass.getCanonicalName());
-  }
-
-  private MessagingSettings getMessagingSettings() {
-    var client = new StreamPipesClientResolver().makeStreamPipesClientInstance();
-    return client.adminApi().getMessagingSettings();
+  private SendToBrokerAdapterSink getAdapterSink(AdapterDescription adapterDescription) {
+    return new SendToBrokerAdapterSink(adapterDescription);
```
Comment (bossenti) [inline streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/AdapterPipelineGenerator.java:43]: Great improvement 👍🏼

File: streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/preprocessing/elements/SendToBrokerAdapterSink.java (hunk 2 above)
Comment (dominikriemer) [reply streampipes-extensions-management/src/main/java/org/apache/streampipes/extensions/management/connect/adapter/preprocessing/elements/SendToBrokerAdapterSink.java:91]: see above, we don't support Kubernetes as development environment currently

**Review by bossenti — APPROVED** [review]

**Review by tenthe — APPROVED** [review]

**Pull request merged**